### PR TITLE
#216 fixes the lack of logging seen in the portieris output for decisions

### PR DESCRIPTION
**Pull request by guysports**
Uses a global decisions map for image decisions across multiple containers, and a local denials map returned from the getPatchesFromContainers and appends to the global decisions map.

Then outputs the image:tag keys of the decisions map in the Allow and Deny decision branches

### CODE REVIEW

**Comment by jhart1685** [comment]
@guysports - thanks for the quick action
I pulled down your code and ran it.. It looks better, but I do not get the digest (or tag) in the log lines. Here's an example of a deny (same occurs for allow). I presume that we don't have the digest if Portieris hasn't mutated the image name.
```
portieris-5cd755854-f82wr portieris I1125 10:50:58.592025       1 controller.go:125] Deny for images [image:digest] [uk.icr.io/jhart/volports:]
```

**Comment by guysports** [comment]
@jhart1685 
Regarding the tag, @sjhx asked for it to be removed in my original PR as tags are mutable https://github.com/IBM/portieris/pull/208#pullrequestreview-531902588

Regarding the digest, the idea was that an entry in the decisions map is created for every container passed to the `getPatchesForContainers()` function of the form image:digest, regardless of whether it is mutated or not. However, I see that in `image.NewReference(container.Image)` from which we take the digest, it's only set if the image name has the sha256:digest included
https://github.com/IBM/portieris/blob/9973a31d52a4552d8d3b9cc7c44ca43df55b26f4/helpers/image/image.go#L44

So from looking at a container in one of our clusters
```Containers:
  prometheus:
    Container ID:   containerd://880206a55265fd9c48d81bf35d85368389c08db9b15627e106b7481bde113bc4
    Image:          registry.ng.bluemix.net/mhub/prometheus:710
    Image ID:       registry.ng.bluemix.net/mhub/prometheus@sha256:af01387da7d70f1e891b43ec1d15bb06c9123d691defcc642117f20da6b9c15b
    Port:           9090/TCP
    Host Port:      0/TCP
```
the image ID rather than the image name needs to be passed to `image.NewReference(name)` but it's not immediately obvious to me from the kubernetes doc how the imageID is obtained.

**Comment by guysports** [comment]
@jhart1685 
Investigating further, because the image hasn't been pulled at the point portieris is run, the digest isn't available until portieris obtains it with `c.Enforcer.DigestByPolicy(namespace, img, credentialCandidates, containerPolicy)` so a compromise is to log a decision with the `image:tag` information if a Deny decision is reached before the digest is obtained, or with the `image:digest` if it's an Allow decision or the Deny decision is reached after the digest is obtained

I have pushed a commit to reflect this

**Review by jhart1685 — APPROVED** [review]
This looks like a good compromise to me, thanks. Will ask one of the maintainers to have a quick look too before merging.

**Pull request merged**
